### ISSUE TITLE
Proposal: revamp states and colors

### ISSUE THREAD

**Issue by bakhtiyarneyman** [greshake/i3status-rust#1208]
We have 5 states in the increasing order of priority: Good, Idle, Info, Warning, Critical, and they are typically assigned colors Green, Black, Blue, Yellow, Red. This is confusing for a couple of reasons.
* Blue doesn't lie on the spectrum between Green and Yellow. In vast majority of color schemes the colors signaling severity range from Green to Yellow to Red. There's simply no Blue. I am constantly getting confused about what Blue means. 
* Green is sometimes used to signal quantitative information about something being especially good (e.g. temperature lower than what it's expected to be for idle), and sometimes it is used to signal a qualitative information (e.g. that a bluetooth mouse is connected).
* For battery (despite the default thresholds for good and info being the same, essentially hiding the idle range), the states for discharging battery are Critical, Warning, Idle and Good, and is Good for charging battery.  

We can fix these inconsistencies. I propose to keep the color assignments to the statuses, but organize statuses differently:
````
enum Status {
  // Black. This is the default, when everything is as expected. E.g. CPU load under 10%, temperature
  // under 40 degrees, plugged battery that is fully charged so not charging, volume under 70%, 
  // wifi connected and utilized under 1 MB/s, time, bluetooth device not connected.
  Idle,
  // Green. Slight concern. E.g. CPU load between 10% and 20%, battery unplugged and between 70% and 
  // 100%, wifi connected and utilized at between 1MB/s and 2MB/s, temperature between 40 and 60 
  // degrees, volume between 70% and 100%.
  Good,
  // Yellow. Moderate concern. Same as above but with higher/lower ranges depending on context.
  Warning, 
  // Red. High concern. Same as above, but with even higher/lower ranges depending on context.
  Critical, 
  // Blue. No concern, but represents binary condition that user might be interesting to know 
  // about. E.g. connected to a bluetooth device.
  Info, 
  // Magenta. No concern, but represents that there is a transient condition. E.g. connecting to a 
  // bluetooth device, battery plugged and charging.
  Transient,
}
````
Thus there would be two axes: Black-Green-Yellow-Red for problems and quantitative stuff and Black-Magenta-Blue for status and qualitative stuff. 

This proposal reserves Cyan for future use. The biggest problem is that the proposal will need to break current conventions which might surprise people. Hopefully it will be worth it.

Please let me know what you think. If this proposal gets traction I am intending to implement it.

**Comment by GladOSkar** [comment]
I'm also not happy with the inconsistencies regarding colors in many blocks.

Your suggestions seem generally reasonable, my only nit would be that "Good" to me does not indicate "Slight concern" but rather "Especially good" state, for example when the battery is charging or a device connected. In other words, the "default" state for everything should be all Idle except for deviations towards "better" (Good), "worse" (Warning/Critical) or "notification" (Info/Transient). This would mean that *most* blocks don't have a "Good" state but rather top/bottom out at Idle, which seems reasonable to me.

But either way sounds good to me. Decision is up to @ammgws though

**Comment by bakhtiyarneyman** [comment]
Thanks, Oskar! You are right, `temperature` block uses the convention for Good as in "especially good", and many blocks (e.g. cpu, memory, disk space) skip Good altogether. For slightly concerning they are using Info (which is blue). It's a valid point in the design space of course. I edited my original proposal in the light of this view. I still prefer "green-yellow-read" to "blue-yellow-red" for signaling concern, but of course multiple opinions might exist.

It's my personal preference not to be distracted by the "especially good" condition. I don't see much benefit from knowing that the temperature is unusually low right now. Maybe there are some other scenarios where it's more useful?

Battery uses green to signal that the charge is above certain threshold. Because it's still discharging, to me that's more in the "slight concern" territory, rather than "especially good". 

Connected device can indeed be viewed as "especially good" condition, although I would argue that it's also more of a neutral thing. Imagine bluetooth headset connected to the computer. It seems reasonable to say that the preference to use headset vs speakers is up to the user and their desires in that particular moment.

FWIW I think I can solve almost all my problems by simply
* overriding `Info` color to be green globally
* overriding `Good` color to be blue globally (or at least for the bluetooth devices)
* overriding `Good` color to be black, setting the Good threshold to 100 for battery.
* setting `Good` threshold to be -273 for temperature.

That will leave me with black (and not magenta) color for charging, but I guess it's not that bad.

**Comment by GladOSkar** [comment]
i think `Good` = green and `Info` = blue are still the color matchings that make more sense. But i also think we can ditch both `Good` and `Info` states from quantitative values like temperature, `Idle`->`Warning`->`Critical` is absolutely enough differentiation. Then i'd keep `Good` only for things like Battery charging/fully charged of device connected (maybe with an option to leave those at idle if users don't want the distraction) and `Info` for notification-type things.

**Comment by ammgws** [comment]
To be honest I never really understood the states either (the names anyway). I've basically disabled all colours in my config too, so blocks only come up red when there's something worth looking at. Would be interested to see what others think / are doing with the current states/colours.

**Comment by GladOSkar** [comment]
It's similar for me but i like to have a little color there, mostly because it looks nice i guess. So it's mostly gray with a few colors:

![image](https://user-images.githubusercontent.com/9497832/117481265-08408000-af63-11eb-8c93-f487f83345a4.png)

**Comment by bakhtiyarneyman** [comment]
Closing because it didn't get traction.

**Comment by ammgws** [comment]
I think it's still worth looking into. It might be even easier to judge if someone made a PR so we can experience the new behaviour visually. Personally I think I would be happy with just 3 states: Idle (black), Warning (yellow) and Critical (red), but it might make sense to write out a complete list of which block is using what state and for what purpose in order to see if that is feasible.

**Comment by MaxVerevkin** [comment]
> Personally I think I would be happy with just 3 states: Idle (black), Warning (yellow) and Critical (red),

I think we also _do_ need Info(blue/green/cyan) to indicate connected bluetooth or playing music

**Comment by hw0lff** [comment]
I think that the assignment of states to colors needs a reevaluation. Not everything has a certain number of states. For example network interfaces. They can be connected, disconnected, disabled, non-existent (if there is an adapter involved), connect but not having an IP address assigned to them. The signal strength of WiFi interfaces and bandwidth would need thresholds, preferably configurable. 
I don't think it is possible or even desirable to be limited to 5 states and colors.  I would propose that the 5 (or 6, as mentioned in the first post) states are only the base instances of states. In a way that they can be used by blocks but more importantly also be easily expanded by blocks to add more states that don't fit the standard mapping. And therefore provide more fitting color assignments that are also configurable by the end user. 

tldr: Rework the states so that the list of states can be easily extended and edited inside and specifically for a block. And color assignments to a block specific state can be reconfigured by the user with theme-overrides.
